### ISSUE TITLE
Users can post in read-only channels

### ISSUE THREAD

**Issue by jrainville** [status-im/status-desktop#13792]
# Bug Report

## Description

I created a read-only channel. Only owners, TMs and admins should be able to post to it, but somehow normal users sitll can.

There is a UI bug where the chat input is not correctly disabled, but this is for the backend. We let the messages through and they should be blocked.

The issue seems like that user was put in the member list of the channel even though he shouldn't.

## Steps to reproduce

1. Have a community
2. Create a new channel
3. add a read-only channel 
![image](https://github.com/status-im/status-desktop/assets/11926403/a191a01f-df52-4ece-b982-cab0df2904ec)
4. Open another app
5. Create a new account
6. Invite that new account to the community
7. New account joins
8. See that the new account is in the member list of the read-only channel
9. New account can post

normal-guy here is not supposed to be in the list
![image](https://github.com/status-im/status-desktop/assets/11926403/36ea58cc-4bd2-4e80-9409-098dc01d9289)

#### Expected behavior

Only owners, admins and TMs are present on the list of members of a read-only channel and only them can post

#### Actual behavior

A normal user is present in the list

### Additional Information

- Status desktop version: master `c9fd4ebceea3e10b19633aae4d5f4bdf2c10a9fd`
- Operating System: Ubuntu 20

**Comment by jrainville** [comment]
Could be related to https://github.com/status-im/status-desktop/issues/13647 cc @osmaczko

**Comment by osmaczko** [comment]
I've just checked, and it appears that it was never implemented to be blocked on the backend. In `CanPost`, we only check if the member belongs to the members list.

We need to extend `CommunityMember` with a `permissions` field (CAN_VIEW, CAN_POST) and ensure that the control node shares this within `CommunityDescription`.

**Comment by jrainville** [comment]
Yeah, that's why I mentioned the member list in the issue. If we "fix" the member list to not include members that are not admins, then it should just work.

Maybe it would look weird that other users don,t appear in the member list though? Anyway, I think it would be the easiest fix for now.

**Comment by jrainville** [comment]
Ok makes sense. Then yes, we should go with your solution

**Comment by igor-sirotin** [comment]
> We need to extend CommunityMember with a permissions field (CAN_VIEW, CAN_POST) and ensure that the control node shares this within CommunityDescription.

I've done this in my branch here: https://github.com/status-im/status-go/pull/4820

**Comment by caybro** [comment]
Related UI issue: https://github.com/status-im/status-desktop/issues/13777 I'm working on

**Comment by jrainville** [comment]
> > We need to extend CommunityMember with a permissions field (CAN_VIEW, CAN_POST) and ensure that the control node shares this within CommunityDescription.
> 
> I've done this in my branch here: [status-im/status-go#4820](https://github.com/status-im/status-go/pull/4820)

So does this issue still need some work or the PR you linked fixed it @igor-sirotin ?

**Comment by iurimatias** [comment]
moving to 2.29 due to lack of space in this milestone, but we might move it back during the stabilization period

**Comment by igor-sirotin** [comment]
> > > We need to extend CommunityMember with a permissions field (CAN_VIEW, CAN_POST) and ensure that the control node shares this within CommunityDescription.
> > 
> > 
> > I've done this in my branch here: [status-im/status-go#4820](https://github.com/status-im/status-go/pull/4820)
> 
> So does this issue still need some work or the PR you linked fixed it @igor-sirotin ?

The PR is linked, so `CAN_VIEW`/`CAN_POST` channel roles can be already used. If needed, ofc.

**Comment by jrainville** [comment]
Moving to 2.29 since there won't be the time to do

**Comment by jrainville** [comment]
> To add more info:
> 
> 1. The code that answer a community request to join has this bug
> 2. The code that periodically checks for permissions is correct
> 
> So the behaviour is that once a user requests to join a commnuity, initially will be able to post on channels who he doesn't have access to. Once 2 is run (after one hour or something), it eventually fixes itself.
> 
> Just helpful to replicate and avoid confusion.

Just an FYI, I pushed a fix lately to the issue where the member role wasn't updated correctly on re-evaluate: https://github.com/status-im/status-go/pull/4900

I think that's what you meant when saying the periodic check does it correctly, but just giving as much info as possible :D

**Comment by cammellos** [comment]
> I think that's what you meant when saying the periodic check does it correctly, but just giving as much info as possible

It's a different issue, it's easier to explain once I push a fix, since I forgot exactly how it works, but basically the two checks are inconsistent in how the interpret the multiple permissions on a single channel, I'll ping you as soon as it's out

**Comment by jrainville** [comment]
This is now fixed
